sdk/trace/span.go

```diff
@@ -426,6 +427,15 @@ func (s *recordingSpan) End(options ...trace.SpanEndOption) {
 	s.mu.Unlock()
 
 	sps := s.tracer.provider.getSpanProcessors()
+	for _, sp := range sps {
+		if oesp, ok := sp.sp.(x.OnEndingSpanProcessor); ok {
+			oesp.OnEnding(s)
+		}
+	}
+	s.mu.Lock()
```

Comment:
This acquires the lock twice. Can this be avoided instead?

Reply:
Not really, as we need to set the end time, and later on, mark the span as having ended.

Reply:
I've changed to only lock once if there are no onending processors.